src/Microsoft.ML.Data/Transforms/DropSlotsTransform.cs

```diff
@@ -190,98 +240,81 @@ private static VersionInfo GetVersionInfo()
                 verReadableCur: 0x00010001,
                 verWeCanReadBack: 0x00010001,
                 loaderSignature: LoaderSignature,
-                loaderAssemblyName: typeof(DropSlotsTransform).Assembly.FullName);
+                loaderAssemblyName: typeof(SlotsDroppingTransformer).Assembly.FullName);
         }
 
-        private const string RegistrationName = "DropSlots";
-
         /// <summary>
-        /// Extra information for each column (in addition to ColumnInfo).
+        /// Initializes a new <see cref="SlotsDroppingTransformer"/> object.
         /// </summary>
-        private sealed class ColInfoEx
+        /// <param name="env">The environment to use.</param>
+        /// <param name="input">Name of the input column.</param>
+        /// <param name="output">Name of the column resulting from the transformation of <paramref name="input"/>. Null means <paramref name="input"/> is replaced. </param>
+        /// <param name="min">Specifies the lower bound of the range of slots to be dropped. The lower bound is inclusive. </param>
+        /// <param name="max">Specifies the upper bound of the range of slots to be dropped. The upper bound is exclusive.</param>
+        public SlotsDroppingTransformer(IHostEnvironment env, string input, string output = null, int min = default, int? max = null)
+            : this(env, new ColumnInfo(input, output, (min, max)))
         {
-            public readonly SlotDropper SlotDropper;
-            // Track if all the slots of the column are to be dropped.
-            public readonly bool Suppressed;
-            public readonly ColumnType TypeDst;
-            public readonly int[] CategoricalRanges;
-
-            public ColInfoEx(SlotDropper slotDropper, bool suppressed, ColumnType typeDst, int[] categoricalRanges)
-            {
-                Contracts.AssertValue(slotDropper);
-                SlotDropper = slotDropper;
-                Suppressed = suppressed;
-                TypeDst = typeDst;
-                CategoricalRanges = categoricalRanges;
-            }
         }
 
-        private readonly ColInfoEx[] _exes;
-
         /// <summary>
-        /// Public constructor corresponding to SignatureDataTransform.
+        /// Initializes a new <see cref="SlotsDroppingTransformer"/> object.
         /// </summary>
-        public DropSlotsTransform(IHostEnvironment env, Arguments args, IDataView input)
-            : base(Contracts.CheckRef(env, nameof(env)), RegistrationName, env.CheckRef(args, nameof(args)).Column, input, null)
+        /// <param name="env">The environment to use.</param>
+        /// <param name="columns">Specifies the ranges of slots to drop for each column pair.</param>
+        public SlotsDroppingTransformer(IHostEnvironment env, params ColumnInfo[] columns)
```

Comment:
>public [](start = 8, length = 6)

private/internal ?

Reply:
Since SlotsDroppingTransformer is not trainable, it should have public constructor, so this should be fine.

---
In reply to: [235558026](https://github.com/dotnet/machinelearning/pull/1683#discussion_r235558026) [](ancestors = 235558026)